kraken-common/src/kraken/common/_runner.py

```diff
@@ -109,7 +109,10 @@ def execute_script(self, script: Path, scope: Dict[str, Any]) -> None:
         module = types.ModuleType(str(script.parent))
         module.__file__ = str(script)
 
-        code = compile(script.read_text(), script, "exec")
+        try:
+            code = compile(script.read_text(), script, "exec")
+        except BaseException as err:
+            raise ValueError("Script failed to compile the build script") from err
```

Comment:
This doesn't feel quite right. For one, because it catches any and all exceptions when really we probably just want to catch `SyntaxError`s. And what benefit does the reraise as a `ValueError` give us?

Reply:
Agree with catching all errors being a bit of a nuclear option here, but assumed as it was only 'compiling' and not running the python code, that this would be OK. Catching `Syntax Error` is a good suggestion, but perhaps also `Assertion`, `Import`, `ModuleNotFound`, `Indentation`, `Type`, `Unicode` and `ZeroDivision` errors would also be good?
The re-raise gives us the cleaner error message to handle later and give to the end-user. It'll get unrolled to give the original error if we want to display it;